t/plugin/limit-count-redis.t

```diff
@@ -209,3 +209,95 @@ passed
 [404, 503, 404, 503, 503]
 --- no_error_log
 [error]
+
```

Comment:
the test cases are not cover redis password. For example, right or wrong password

Reply:
> the test cases are not cover redis password. For example, right or wrong password

`right or wrong` redis password is based on modifing the redis service (maybe in travis ci script? I'm confused with this now). Which may make  other test failed.

Reply:
yes, the redis server installed by travis CI, and only this plugin will use redis now.
IMO you should add test cases for this PR, otherwise we can not sure it works.

Reply:
> yes, the redis server installed by travis CI, and only this plugin will use redis now.
> IMO you should add test cases for this PR, otherwise we can not sure it works.

@moonming  Help needed about how to make the test cases below right.

### request with wrong redis_password
``` bash
$ curl -i 127.0.0.1:80/hello
HTTP/1.1 500 Internal Server Error
Date: Tue, 24 Dec 2019 01:42:49 GMT
Content-Type: text/html
Content-Length: 591
Connection: close
Server: APISIX web server

```

### logs output

``` bash
$ docker-compose logs -f
[error] 41#41: *77 [lua] limit-count.lua:127: phase_fun(): failed to limit req: ERR invalid password, client: 127.0.0.1, server: , request: "GET /hello HTTP/1.1", host: "127.0.0.1"

127.0.0.1 "GET /hello HTTP/1.1" 500 591 0.001 "-" "curl/7.29.0" - - -
```

### test case

```
=== TEST 6: set route, with redis host, port and wrong password
--- config
    location /t {
        content_by_lua_block {
            local t = require("lib.test_admin").test
            local code, body = t('/apisix/admin/routes/1',
                ngx.HTTP_PUT,
                [[{
                    "plugins": {
                        "limit-count": {
                            "count": 2,
                            "time_window": 60,
                            "rejected_code": 503,
                            "key": "remote_addr",
                            "policy": "redis",
                            "redis_host": "127.0.0.1",
                            "redis_port": 6379,
                            "redis_timeout": 1001,
                            "redis_password": "WRONG_foobared"
                        }
                    },
                    "upstream": {
                        "nodes": {
                            "127.0.0.1:1980": 1
                        },
                        "type": "roundrobin"
                    },
                    "uri": "/hello"
                }]]
                )

            if code >= 300 then
                ngx.status = code
            end
            ngx.print(body)
        }
    }
--- request
GET /t
--- response_body
passed
--- no_error_log
[error]



=== TEST 7: up the limit for TEST 6
--- pipelined_requests eval
["GET /hello", "GET /hello", "GET /hello", "GET /hello"]
--- error_code eval
[500, 500, 500, 500]
--- no_error_log
[error]



=== TEST 8: up the limit for TEST 6
--- pipelined_requests eval
["GET /hello1", "GET /hello", "GET /hello2", "GET /hello", "GET /hello"]
--- error_code eval
[404, 500, 404, 500, 500]
--- no_error_log
[error]
```

### travis ci errors:

```
#   Failed test 'TEST 6: set route, with redis host, port and wrong password - response_body - response is expected (repeated req 0, req 0)'
1742#   at /home/travis/build/apache/incubator-apisix/test-nginx/lib/Test/Nginx/Socket.pm line 1589.
1743#          got: 'passed'
1744#     expected: 'passed
1745# '
1746t/plugin/limit-count-redis.t ........... 31/? 
1747#   Failed test 'TEST 7: up the limit for TEST 6 - pattern "[error]" should not match any line in error.log but matches line "2019/12/23 16:04:00 [error] 31631\#31631: *333 [lua] limit-count.lua:127: phase_fun(): failed to limit req: ERR invalid password, client: 127.0.0.1, server: localhost, request: \"GET /hello HTTP/1.1\", host: \"localhost\"" (req 0)
1748# '
1749#   at /home/travis/build/apache/incubator-apisix/test-nginx/lib/Test/Nginx/Socket.pm line 1280.
1750
1751#   Failed test 'TEST 7: up the limit for TEST 6 - status code ok'
1752#   at /home/travis/build/apache/incubator-apisix/test-nginx/lib/Test/Nginx/Socket.pm line 935.
1753#          got: ''
1754#     expected: '500'
1755
1756#   Failed test 'TEST 7: up the limit for TEST 6 - status code ok'
1757#   at /home/travis/build/apache/incubator-apisix/test-nginx/lib/Test/Nginx/Socket.pm line 935.
1758#          got: ''
1759#     expected: '500'
1760
1761#   Failed test 'TEST 7: up the limit for TEST 6 - status code ok'
1762#   at /home/travis/build/apache/incubator-apisix/test-nginx/lib/Test/Nginx/Socket.pm line 935.
1763#          got: ''
1764#     expected: '500'
1765t/plugin/limit-count-redis.t ........... 39/? 
1766#   Failed test 'TEST 8: up the limit for TEST 6 - pattern "[error]" should not match any line in error.log but matches line "2019/12/23 16:04:01 [error] 31683\#31683: *384 [lua] limit-count.lua:127: phase_fun(): failed to limit req: ERR invalid password, client: 127.0.0.1, server: localhost, request: \"GET /hello HTTP/1.1\", host: \"localhost\"" (req 0)
1767# '
1768#   at /home/travis/build/apache/incubator-apisix/test-nginx/lib/Test/Nginx/Socket.pm line 1280.
1769
1770#   Failed test 'TEST 8: up the limit for TEST 6 - status code ok'
1771#   at /home/travis/build/apache/incubator-apisix/test-nginx/lib/Test/Nginx/Socket.pm line 935.
1772#          got: ''
1773#     expected: '404'
1774
1775#   Failed test 'TEST 8: up the limit for TEST 6 - status code ok'
1776#   at /home/travis/build/apache/incubator-apisix/test-nginx/lib/Test/Nginx/Socket.pm line 935.
1777#          got: ''
1778#     expected: '500'
1779
1780#   Failed test 'TEST 8: up the limit for TEST 6 - status code ok'
1781#   at /home/travis/build/apache/incubator-apisix/test-nginx/lib/Test/Nginx/Socket.pm line 935.
1782#          got: ''
1783#     expected: '500'
1784# Looks like you failed 9 tests of 48.
1785t/plugin/limit-count-redis.t ........... Dubious, test returned 9 (wstat 2304, 0x900)
1786Failed 9/48 subtests 
```